citesphere/src/test/java/edu/asu/diging/citesphere/core/util/model/impl/CitationHelperTest.java

```diff
@@ -0,0 +1,350 @@
+        editor1.setFirstName("editor");
+        editor1.setLastName("editor");
+        editor1.setId("1");
+        editorsSet.add(editor1);
+        updatedCitation.setEditors(editorsSet);
+        Set<IPerson> authorsSet = new LinkedHashSet<IPerson>();
+        IPerson author1 = new Person();
+        author1.setFirstName("first");
+        author1.setLastName("last");
+        author1.setId("2");
+        Set<IAffiliation> affiliations = new LinkedHashSet<IAffiliation>();
+        IAffiliation affiliation = new Affiliation();
+        affiliation.setName("Test");
+        affiliations.add(affiliation);
+        author1.setAffiliations(affiliations);
+        authorsSet.add(author1);
+        updatedCitation.setAuthors(authorsSet);
+        Set<ICitationConceptTag> conceptTagsSet = new LinkedHashSet<ICitationConceptTag>();
+        ICitationConceptTag tag1 = new CitationConceptTag();
+        tag1.setConceptName("testc.com");
+        tag1.setConceptUri("www.google.com");
+        tag1.setLocalConceptId("CON1");
+        tag1.setLocalConceptTypeId("CTY1");
+        tag1.setTypeName("forum");
+        tag1.setTypeUri("www.test.com");
+        conceptTagsSet.add(tag1);
+        updatedCitation.setConceptTags(conceptTagsSet);
+    }
+
+    @Test
+    public void test_updateCitation() {
+        ICitation citation;
+        citation = new Citation();
+        citation.setTitle("title");
+        citation.setShortTitle("shortTitle");
+        citation.setDateFreetext("2019");
+        citation.setVolume("1.0");
+        ICitationConcept concept = new CitationConcept();
+        concept.setId("CON1");
+        concept.setOwner(user);
+        concept.setUri("www.google.com");
+        IConceptType type = new ConceptType();
+        type.setId("CTY1");
+        type.setOwner(user);
+        type.setName("forum");
+        type.setUri("www.test.com");
+        Mockito.when(conceptManager.save(any(ICitationConcept.class))).thenReturn(concept);
+        Mockito.when(typeManager.save(any(IConceptType.class))).thenReturn(type);
+        helperToTest.updateCitation(citation, form, user);
+        assertTrue(equalsCitation(citation, updatedCitation));
+    }
+
+    @Test
+    public void test_updateCitation_getByURIAndOwner() {
+        ICitation citation;
+        citation = new Citation();
+        citation.setTitle("title");
+        citation.setShortTitle("shortTitle");
+        citation.setDateFreetext("2019");
+        citation.setVolume("1.0");
+        ICitationConcept concept = new CitationConcept();
+        concept.setId("CON1");
+        concept.setOwner(user);
+        concept.setUri("www.google.com");
+        IConceptType type = new ConceptType();
+        type.setId("CTY1");
+        type.setOwner(user);
+        type.setName("forum");
+        type.setUri("www.test.com");
+        
+        Mockito.when(conceptManager.getByUriAndOwner(Mockito.argThat(new URIMatcher()), any(IUser.class))).thenReturn(concept);
```

Comment:
you only want concept be returned, when the user and the concept uri is given, right? Not for any uri and any user.

same for the typemanager below